src/psyclone/psyad/transformations/__init__.py

```diff
@@ -0,0 +1,42 @@
+# BSD 3-Clause License
+#
+# Copyright (c) 2021, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors: R. W. Ford and A. R. Porter, STFC Daresbury Lab
+
+'''PSyAD transformations package module'''
+
+from psyclone.psyad.transformations.tangentlinear_error import \
+    TangentLinearError
+from psyclone.psyad.transformations.adjoint_trans import AdjointTransformation
+from psyclone.psyad.transformations.assignment_trans import AssignmentTrans
+
+__all__ = ["AssignmentTrans", "TangentLinearError"]
```

Comment:
I missed it out on purpose as I thought it was more of an internal development thing rather than something to expose to the user. Do you disagree?

I would remove the import but it would mean that the transformations that use it would need to include it from the original file. I can change the code to do that if you like and then remove the import here.

Reply:
I think I'd prefer your second suggestion as, as you say, it's a base class. (I thought that a module had to be in the `__all__` list in order to be made available by a package so I must be missing something.)

Reply:
I was missing something (see https://docs.python.org/3/tutorial/modules.html). The `__all__` list in `__init__.py` is only there to specify what `from my.package import *` actually imports. Given that that usage is frowned upon (and we don't employ it), the `__all__` in `__init__.py` is only used by autoapi. Since this then causes errors/warnings about multiply-defined symbols I would suggest we stop adding `__all__` to the `__init__.py` files. (See #1280)

Reply:
The import of `AdjointTransformation` is still there?